src/Dialog/Dialog.test.tsx

```diff
@@ -0,0 +1,184 @@
+                <DialogButton onClick={noop}>Close</DialogButton>
+            </Dialog.Footer>
+        </Dialog>
+    );
+
+    describe('is visible when it', () => {
+        it('is rendered when visible', () => {
+            render(dialog());
+            expect(screen.getByRole('dialog')).toBeInTheDocument();
+            test;
+        });
+
+        it('is not rendered when invisible', () => {
+            render(dialog(false));
+            expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+        });
+    });
+
+    describe('shows the expected content', () => {
+        it('in the title', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Title')).toBeInTheDocument();
+        });
+
+        it('in the body', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Text')).toBeInTheDocument();
+        });
+
+        it('in the footer', () => {
+            render(dialog());
+
+            expect(screen.getByText('Close')).toBeInTheDocument();
+        });
+    });
+});
+
+describe('InfoDialog creator', () => {
+    const dialog = (isVisible = true) => (
+        <InfoDialog isVisible={isVisible} onClose={noop}>
+            Test Body
+        </InfoDialog>
+    );
+
+    describe('shows the expected content', () => {
+        it('in the title', () => {
+            render(dialog());
+
+            expect(screen.getByText('Info')).toBeInTheDocument();
+        });
+
+        it('in the body', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Body')).toBeInTheDocument();
+        });
+
+        it('in the footer', () => {
+            render(dialog());
+
+            expect(screen.getByText('Close')).toBeInTheDocument();
+        });
+    });
+
+    it('invokes the expected action', () => {
+        render(dialog());
+
+        const closeButton = screen.getByText('Close');
+        fireEvent.click(closeButton);
+
+        expect(noop).toHaveBeenCalled();
+    });
+});
+
+describe('ErrorDialog creator', () => {
+    const dialog = (isVisible = true) => (
+        <ErrorDialog isVisible={isVisible} onClose={noop}>
+            Test Body
+        </ErrorDialog>
+    );
+
+    describe('shows the expected content', () => {
+        it('in the title', () => {
+            render(dialog());
+
+            expect(screen.getByText('Error')).toBeInTheDocument();
+        });
+
+        it('in the body', () => {
+            render(dialog());
+
+            expect(screen.getByText('Test Body')).toBeInTheDocument();
+        });
+
+        it('in the footer', () => {
+            render(dialog());
+
+            expect(screen.getByText('Close')).toBeInTheDocument();
+        });
+    });
+
+    it('invokes the expected action', () => {
+        render(dialog());
+
+        const closeButton = screen.getByText('Close');
+        fireEvent.click(closeButton);
+
+        expect(noop).toHaveBeenCalled();
+    });
+});
```

Comment:
Could even consider to just test all this in a single test, would then do it with the above as well. Then you don't need a describe and could just create a test in the body of the file.

Reply:
I combined the `expected content` ones but like the action one being seperate for now.
Will change this if you really want me to.